rust/processor/src/db/common/models/emojicoin_models/models/chat_event.rs

```diff
@@ -0,0 +1,131 @@
+use crate::db::common::models::emojicoin_models::json_types::{StateEvent, TxnInfo};
+use crate::db::common::models::emojicoin_models::utils::micros_to_naive_datetime;
+use crate::db::common::models::emojicoin_models::{enums, json_types::ChatEvent};
+use crate::schema::chat_events;
+use bigdecimal::BigDecimal;
+use field_count::FieldCount;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Deserialize, FieldCount, Identifiable, Insertable, Serialize)]
+#[diesel(primary_key(market_id, market_nonce))]
+#[diesel(table_name = chat_events)]
+pub struct ChatEventModel {
+    // Transaction metadata.
+    pub transaction_version: i64,
+    pub sender: String,
+    pub entry_function: Option<String>,
+    pub transaction_timestamp: chrono::NaiveDateTime,
+
+    // Market and state metadata.
+    pub market_id: i64,
+    pub symbol_bytes: Vec<u8>,
+    pub bump_time: chrono::NaiveDateTime,
+    pub market_nonce: i64,
+    pub trigger: enums::Trigger,
+
+    // Chat event data.
+    pub user: String,
```

Comment:
Possible to use `AccountAddress` as in the `allowlist` model?

Will avoid hassles with sanitation later